lib/SPIRV/OCLToSPIRV.cpp

```diff
@@ -323,8 +323,8 @@ void OCLToSPIRVBase::visitCallInst(CallInst &CI) {
     return;
   }
   if (DemangledName == kOCLBuiltinName::Dot &&
-      (CI.getOperand(0)->getType()->isFloatTy() ||
-       CI.getOperand(1)->getType()->isDoubleTy())) {
+      (!CI.getOperand(0)->getType()->isVectorTy() &&
+       !CI.getOperand(0)->getType()->isIntegerTy())) {
```

Comment:
Wondering if just `CI.getOperand(0)->getType()->isFPOrFPVectorTy()` is sufficient?

Reply:
From what I understand this visitCallDot is meant to translate dot with scalar floating point arguments (but not vectors) into FMul. Because of that FPVectors shouldn't pass this check but instead stay as calls to dot.

Reply:
`isFloatingPointTy()`?

Reply:
Yeah, that might be a good idea